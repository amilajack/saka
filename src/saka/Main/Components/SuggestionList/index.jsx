import 'material-components-web/dist/material-components-web.css';
import 'scss/styles.scss';
import { h, Component } from 'preact';
import Suggestion from './Containers/SuggestionSelector.jsx';

export default ({
  searchString,
  suggestions,
  selectedIndex,
  firstVisibleIndex,
  maxSuggestions,
  onSuggestionClick
}) => (
  <ul className="mdc-list mdc-list--two-line mdc-list--avatar-list two-line-avatar-text-icon-demo list-container">
    {suggestions
      .slice(firstVisibleIndex, firstVisibleIndex + maxSuggestions)
      .map((suggestion, index) => (
        <Suggestion
          suggestion={suggestion}
          searchString={searchString}
          selected={index === selectedIndex}
          index={index}
          onClick={onSuggestionClick}
        />
      ))}
  </ul>
);
