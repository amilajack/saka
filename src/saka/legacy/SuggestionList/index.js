import { h } from 'preact';
import { isURL } from 'lib/url';
import Suggestion from '../Suggestion';
import '@material/list/dist/mdc.list.min.css';
import './style.css';

export default ({ searchText, suggestions, selectedIndex }) => (
  <ul className="mdc-list mdc-list--two-line mdc-list--avatar-list two-line-avatar-text-icon-demo list-container">
    {suggestions.map((suggestion, index) => (
      <Suggestion
        key={index}
        suggestion={suggestion}
        searchText={searchText}
        detail={undefined}
        type={isURL(suggestion) ? 'url' : 'search'}
        selected={index === selectedIndex}
        index={index}
        role="listbox"
      />
    ))}
  </ul>
);
