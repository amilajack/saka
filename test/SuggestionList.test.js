import SuggestionList from '@/saka/Main/Components/SuggestionList/index.jsx';
import { render } from 'preact-testing-library';
import { h } from 'preact';

const MAX_SUGGESTIONS = 6;

describe('SuggestionList component ', () => {
  it('should be empty when no values provided', () => {
    const suggestions = [];

    const searchString = {};

    const { container } = render(
      <SuggestionList
        searchString={searchString}
        suggestions={suggestions}
        selectedIndex={0}
        firstVisibleIndex={0}
        maxSuggestions={5}
      />
    );

    const suggestionList = wrapper.find('.tab-suggestion');
    expect(suggestionList.length).toBe(0);
  });

  it('should display suggestions when provided', () => {
    const suggestions = [
      {
        type: 'tab',
        title: 'lusakasa/saka: Elegant tab search',
        url: 'https://github.com/lusakasa/saka'
      },
      {
        type: 'tab',
        title: 'Google',
        url: 'https://google.com'
      }
    ];

    const searchString = '';

    const SuggestionListRender = render(
      <SuggestionList
        searchString={searchString}
        suggestions={suggestions}
        selectedIndex={0}
        firstVisibleIndex={0}
        maxSuggestions={MAX_SUGGESTIONS}
        onSuggestionClick={() => {}}
      />
    );

    const suggestionList = SuggestionListRender.find('.tab-suggestion');
    expect(suggestionList.length).toBe(2);

    suggestions.map((suggestion, index) => {
      const renderedSuggestion = suggestionList.at(index).attrs();
      expect(renderedSuggestion.type).toBe(suggestion.type);
      expect(renderedSuggestion.title).toBe(suggestion.title);
      expect(renderedSuggestion.url).toBe(suggestion.url);
    });
  });
});
