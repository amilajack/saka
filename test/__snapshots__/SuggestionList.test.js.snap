// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionList component  should be empty when no values provided 1`] = `
<div>
  <ul
    class="mdc-list mdc-list--two-line mdc-list--avatar-list two-line-avatar-text-icon-demo list-container"
  />
</div>
`;
